From d02bdfc32e748d3763d639cf291bffff4b121df2 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@gmail.com>
Date: Sun, 3 Nov 2013 08:50:03 +0100
Subject: [PATCH] Cache configure result of getcwd(NULL,0) test.

This is the last configure test that must be cached for a proper
cross compilation.
---
 configure.ac | 22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3ce8e8e..fbe6c57 100644
--- configure.ac
+++ configure.ac
@@ -58,16 +58,20 @@ KYUA_MEMORY
 AC_CHECK_FUNCS([putenv setenv unsetenv])
 
 
-AC_MSG_CHECKING([whether getcwd(NULL, 0) works])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <stdlib.h>
+AC_CACHE_CHECK(
+	[whether getcwd(NULL, 0) works],
+	[kyua_cv_getcwd_null_0_works], [
+	AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <stdlib.h>
 #include <unistd.h>],
-    [char *cwd = getcwd(NULL, 0);
-     return (cwd != NULL) ? EXIT_SUCCESS : EXIT_FAILURE;])],
-    [AC_MSG_RESULT(yes)
-     AC_DEFINE([HAVE_GETCWD_DYN], [1],
-               [Define to 1 if getcwd(NULL, 0) works])],
-    [AC_MSG_RESULT(no)])
-
+	[char *cwd = getcwd(NULL, 0);
+	return (cwd != NULL) ? EXIT_SUCCESS : EXIT_FAILURE;])],
+	[kyua_cv_getcwd_null_0_works=yes],
+	[kyua_cv_getcwd_null_0_works=no])
+])
+if test "${kyua_cv_getcwd_null_0_works}" = yes; then
+	AC_DEFINE([HAVE_GETCWD_DYN], [1],
+		[Define to 1 if getcwd(NULL, 0) works])
+fi
 
 AC_PROG_RANLIB
 
-- 
1.8.4.1

